src/jit/gentree.h

```diff
@@ -2399,12 +2399,23 @@ struct ReturnTypeDesc
 public:
     ReturnTypeDesc()
     {
+#ifdef _TARGET_X86_
+        m_regType0 = TYP_INT;
+        m_regType1 = TYP_INT;
+
+#ifdef DEBUG
+        m_inited = true;
+#endif
+
+#else
```

Comment:
Look at the TODO-AllArch comment on line 2528.
In general, I don't like having off-by-one behavior when it's not necessary, but most especially when it is temporary.
I would not recommend having a Reinitialize function when a node is morphed - it should instead have a "change" or "morph" verb in its name.